src/snowflake/cli/_plugins/nativeapp/entities/application.py

```diff
@@ -476,13 +717,18 @@ def create_or_upgrade_app(
                         return
 
                     except ProgrammingError as err:
-                        if err.errno not in UPGRADE_RESTRICTION_CODES:
-                            generic_sql_error_handler(err=err)
-                        else:  # The existing application object was created from a different process.
+                        if err.errno == CANNOT_DISABLE_MANDATORY_TELEMETRY:
+                            event_sharing.event_sharing_error(
+                                "Could not disable telemetry event sharing for the application because it contains mandatory events. Please set 'share_mandatory_events' to true in the application telemetry section of the project definition file.",
+                                err,
+                            )
+                        elif err.errno in UPGRADE_RESTRICTION_CODES:
```

Comment:
it's hard to tell which sql calls above could throw these errors. there are quite a few queries being executed in the try block.

Reply:
I will defer this to later PR - where I move the rest of the logic to the Facade, then it will be clearer what's causing this.